sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/DLPReidentifyText.java

```diff
@@ -0,0 +1,206 @@
+
+  @Nullable
+  public abstract InspectConfig inspectConfig();
+
+  @Nullable
+  public abstract DeidentifyConfig reidentifyConfig();
+
+  @Nullable
+  public abstract String csvDelimiter();
+
+  @Nullable
+  public abstract PCollectionView<List<String>> csvHeaders();
+
+  public abstract Integer batchSize();
+
+  public abstract String projectId();
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setInspectTemplateName(String inspectTemplateName);
+
+    public abstract Builder setInspectConfig(InspectConfig inspectConfig);
+
+    public abstract Builder setReidentifyConfig(DeidentifyConfig deidentifyConfig);
+
+    public abstract Builder setReidentifyTemplateName(String deidentifyTemplateName);
+
+    public abstract Builder setBatchSize(Integer batchSize);
+
+    public abstract Builder setCsvHeaders(PCollectionView<List<String>> csvHeaders);
+
+    public abstract Builder setCsvDelimiter(String delimiter);
+
+    public abstract Builder setProjectId(String projectId);
+
+    public abstract DLPReidentifyText build();
+  }
+
+  public static DLPReidentifyText.Builder newBuilder() {
+    return new AutoValue_DLPReidentifyText.Builder();
+  }
+
+  @Override
+  public PCollection<KV<String, ReidentifyContentResponse>> expand(
+      PCollection<KV<String, String>> input) {
+    return input
+        .apply(ParDo.of(new MapStringToDlpRow(csvDelimiter())))
+        .apply("Batch Contents", ParDo.of(new BatchRequestForDLP(batchSize())))
+        .apply(
+            "DLPDeidentify",
+            ParDo.of(
+                new ReidentifyText(
+                    projectId(),
+                    inspectTemplateName(),
+                    reidentifyTemplateName(),
+                    inspectConfig(),
+                    reidentifyConfig(),
+                    csvHeaders())));
+  }
+
+  public static class ReidentifyText
+      extends DoFn<KV<String, Iterable<Table.Row>>, KV<String, ReidentifyContentResponse>> {
+    private final String projectId;
+    private final String inspectTemplateName;
+    private final String reidentifyTemplateName;
+    private final InspectConfig inspectConfig;
+    private final DeidentifyConfig reidentifyConfig;
+    private transient ReidentifyContentRequest.Builder requestBuilder;
+    private final PCollectionView<List<String>> csvHeader;
+
+    @Setup
+    public void setup() throws IOException {
+      requestBuilder =
+          ReidentifyContentRequest.newBuilder().setParent(ProjectName.of(projectId).toString());
+      if (inspectTemplateName != null) {
+        requestBuilder.setInspectTemplateName(inspectTemplateName);
+      }
+      if (inspectConfig != null) {
+        requestBuilder.setInspectConfig(inspectConfig);
+      }
+      if (inspectConfig == null && inspectTemplateName == null) {
+        throw new IllegalArgumentException(
+            "Either inspectConfig or inspectTemplateName need to be set!");
```

Comment:
for re-id inspect template can be optional. only re-id template is required. we should just check if inspect template passed or not. if not take it as valid request.

Reply:
Thanks!